rsc/sms.properties

```diff
@@ -1,11 +1,38 @@
-# Configuration file for SMS Sender
-# MODE = GSM (COM PORT) or HTTP (PROVIDER)
-# GATEWAY = Skebby or empty
-# TIMEOUT = Timeout for the HTTP request
-# LOOP = loop time in seconds
-# ICC = International Calling Code
-MODE = GSM
-GATEWAY = Skebby
-TIMEOUT = 3000
-LOOP = 15
-ICC = +39
+##################################################################
+# Global configuration
+##################################################################
+# use: gsm-gateway-service || skebby-gateway-service || textbelt-gateway-service
+sms.gateway=textbelt-gateway-service
+sms.gateway.thread.timeout=3000
+sms.gateway.thread.loop=15
+sms.gateway.thread.icc=+39
+
+
+##################################################################
+# GSM configuration
+##################################################################
+gsm-gateway-service.port=COM6
+gsm-gateway-service.driver-name=com.sun.comm.Win32Driver
+gsm-gateway-service.csmp=AT+CSMP\=17,167,0,0\r
+gsm-gateway-service.cmgf=AT+CMGF\=1\r
+gsm-gateway-service.gmm=AT+GMM\r\n 
+gsm-gateway-service.cmgs=AT+CMGS\="
+
+
+##################################################################
+# Skebby configuration
+##################################################################
+# first we will try to use token, if null, then we will try to login with the provided username and password and retrieve token
```

Comment:
where is token here?

Reply:
I took a look to the documentation of skebby. I couldn't test Skebby (they didn't approved my registration, perhaps, for two times with different accounts), but I think that USER_KEY and ACCESS_TOKEN are costants, so that they will never change for a user (there are certainly exceptions). I don't know what could be the value of USER_KEY, but I think that it will never change. I added two new properties USER_KEY and ACCESS_TOKEN. If they exists both, then every time we will try to send sms with Skebby, we will load the authentication informations from a properties file and send sms with the values from properties. I don't know how to retrieve the USER_KEY without make an http request (perhaps we can find the USER_KEY in the user profile?).